core/chains/evm/gas/fixed_price_estimator.go

```diff
@@ -43,8 +47,12 @@ type fixedPriceEstimatorBlockHistoryConfig interface {
 
 // NewFixedPriceEstimator returns a new "FixedPrice" estimator which will
 // always use the config default values for gas prices and limits
-func NewFixedPriceEstimator(cfg fixedPriceEstimatorConfig, bhCfg fixedPriceEstimatorBlockHistoryConfig, lggr logger.Logger) EvmEstimator {
-	return &fixedPriceEstimator{cfg, bhCfg, logger.Sugared(logger.Named(lggr, "FixedPriceEstimator"))}
+func NewFixedPriceEstimator(cfg fixedPriceEstimatorConfig, ethClient evmclient.Client, bhCfg fixedPriceEstimatorBlockHistoryConfig, lggr logger.Logger, chainType config.ChainType) EvmEstimator {
+	var l1Oracle rollups.L1Oracle
+	if rollups.IsRollupWithL1Support(chainType) {
+		l1Oracle = rollups.NewL1GasOracle(lggr, ethClient, chainType)
```

Comment:
Something that I don't quite like with the L1Oracle being created inside the Estimators is that it forces a client object to be passed. So in FixedPriceEstimator's case, you pass an `ethClient` that isn't actually used by the component, it' just needed for L1Oracle initialization. I'll review the rest, and come back with a final comment on this.

Reply:
Not sure if we can get around that since we wanted l1Oracles to be embedded in estimators after this change but I understand the hesitation. Open to suggestions though if you have any after reading the rest of the PR.